File: luogu/p3717.cpp
#include <iostream>
#include <tuple>
#include <vector>
using namespace std;
int main() {
    int n, m, r;
    cin >> n >> m >> r;
    vector<tuple<int, int>> lst;
    int i;
    while (m--) {
        int a, b;
        cin >> a >> b;
        lst.push_back(make_tuple(a, b));
    }
    int count = 0;
    for (int x = 1; x <= n; x++) {
        for (int y = 1; y <= n; y++) {
            for (auto t : lst) {
                auto [xm, ym] = t;
                if ((x - xm) * (x - xm) + (y - ym) * (y - ym) <= r * r) {
                    count++;
                    break;
                }
            }
        }
    }
    cout << count;
    return 0;
}

File: luogu/p3717.py
n,m,r=map(int,input().split())
lst=[]
for _ in range(m):
    x,y=map(int,input().split())
    lst.append((x,y))
count=0
for x in range(1,n+1):
    for y in range(1,n+1):
        for xm,ym in lst:
            if (x-xm)**2+(y-ym)**2<=r**2:
                count+=1
                break
print(count)